src/instruments/src/SD/Pages/Eng/Eng.tsx

```diff
@@ -0,0 +1,312 @@
+
+            if (displayedEngineOilPressure < OIL_PSI_LOW_LIMIT && n2Percent > 75) {
+                setPressureBelowLow(true);
+            }
+
+            if (pressureBelowLow && displayedEngineOilPressure > OIL_PSI_LOW_LIMIT + 2) {
+                setPressureBelowLow(true);
+            }
+
+            if (pressureAboveHigh || pressureBelowLow) {
+                setShouldPressurePulse(true);
+            } else {
+                setShouldPressurePulse(false);
+            }
+
+            if (displayedEngineOilQuantity <= OIL_QTY_LOW_ADVISORY) {
+                setQuantityAtOrBelowLow(true);
+            }
+
+            if (quantityAtOrBelowLow && displayedEngineOilQuantity >= OIL_QTY_LOW_ADVISORY + 2) {
+                setQuantityAtOrBelowLow(false);
+            }
+        } else {
+            setShouldPressurePulse(false);
+            setShouldQuantityPulse(false);
+        }
+
+        if (displayedEngineOilPressure <= OIL_PSI_VLOW_LIMIT) {
+            setPsiNeedleRed(true);
+        }
+        if (psiNeedleRed && displayedEngineOilPressure >= OIL_PSI_VLOW_LIMIT + 0.5) {
+            setPsiNeedleRed(false);
+        }
+    }, [flightPhase, engineOilQuantity, engineOilPressure]);
+
+    useEffect(() => {
+        if (displayedEngineOilTemperature >= OIL_TEMP_HIGH_ADVISORY) {
+            setShouldTemperaturePulse(true);
+        } else if (!tempBeenAboveAdvisory) {
+            setShouldTemperaturePulse(false);
+        }
+
+        if (displayedEngineOilTemperature >= OIL_TEMP_VHIGH_LIMIT) {
+            setTempAmber(true);
+        } else {
+            setTempAmber(false);
+        }
+
+        if (engineOilTemperature > OIL_TEMP_HIGH_ADVISORY) {
+            setTimeout(() => {
+                if (engineOilTemperature > OIL_TEMP_HIGH_ADVISORY) {
+                    setTempBeenAboveAdvisory(true);
+                }
+            }, 900_000);
+        } else {
+            clearTimeout();
+            setTempBeenAboveAdvisory(false);
+        }
+
+        return () => clearTimeout();
+    }, [engineOilTemperature]);
+
+    useEffect(() => {
+        if (tempBeenAboveAdvisory) {
+            setTempAmber(true);
+        }
+    }, [tempBeenAboveAdvisory]);
+
+    useEffect(() => {
+        if (n2Percent >= 50) {
+            setIsValveOpen(false);
+        }
+    }, [n2Percent]);
+
+    useEffect(() => {
+        if (isEngineStarting && n2Percent < 50 && engSelectorPosition === 2) {
+            setTimeout(() => setIsValveOpen(true), 1200);
+        } else {
+            setTimeout(() => setIsValveOpen(false), 1200);
+        }
+
+        return () => clearTimeout();
+    }, [isEngineStarting, engSelectorPosition]);
+
+    useEffect(() => {
+        if (isEngineStarting && n2Igniting && n2Percent > 18 && n2Percent < 55) {
+            setShowIgniter(true);
+        } else {
+            setShowIgniter(false);
+        }
+    }, [isEngineStarting, n2Igniting, n2Percent]);
+
+    return (
+        <SvgGroup x={x} y={y}>
```

Comment:
**issue:** All gauges are in the same `EngineColumn` component.

**suggestion:** Extract every individual gauge to a separate component.